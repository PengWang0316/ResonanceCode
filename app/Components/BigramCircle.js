import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/BigramCircle.module.css';
import growthImage from '../imgs/growth.png';
import conceptionImage from '../imgs/conception.png';
import seedingImage from '../imgs/seeding.png';
import maturationImage from '../imgs/maturation.png';

const BigramCircle = ({ circlePosition, linePoint, isBrown }) => {
  // Start to calculate styles for divs
  let {
    firstDiv, secondDiv, thirdDiv, fourthDiv
  } = styles;
  switch (linePoint) {
    case 1:
      firstDiv = `${firstDiv} ${styles.emphasizedZero}`;
      break;
    case 3:
      fourthDiv = `${fourthDiv} ${styles.emphasizedOne}`;
      break;
    case 2:
      thirdDiv = `${thirdDiv} ${styles.emphasizedTwo}`;
      firstDiv = `${firstDiv} ${styles.cancelBottomBorder}`;
      fourthDiv = `${fourthDiv} ${styles.cancelLeftBorder}`;
      break;
    default:
      firstDiv = `${firstDiv} ${styles.cancelRightBorder}`;
      fourthDiv = `${fourthDiv} ${styles.cancelTopBorder}`;
      secondDiv = `${secondDiv} ${styles.emphasizedThree}`;
      break;
  }
  if (isBrown) {
    firstDiv = `${firstDiv} ${styles.brownBorder}`;
    secondDiv = `${secondDiv} ${styles.brownBorder}`;
    thirdDiv = `${thirdDiv} ${styles.brownBorder}`;
    fourthDiv = `${fourthDiv} ${styles.brownBorder}`;
  }
  return (
    <div className={isBrown ? `${styles[`circle-${circlePosition}`]} ${styles.brownBorder}` : styles[`circle-${circlePosition}`]}>
      <div className={`${styles.internalDiv} d-flex`}>
        <div className={firstDiv}>
          <img style={{ opacity: linePoint === 1 ? 1 : 0.4 }} src={growthImage} alt="growth" />
        </div>
        <div className={secondDiv}>
          <img style={{ opacity: linePoint === 0 ? 1 : 0.4 }} src={conceptionImage} alt="conception" />
        </div>
      </div>
      <div className={`${styles.internalDiv} d-flex`}>
        <div className={thirdDiv}>
          <img style={{ opacity: linePoint === 2 ? 1 : 0.4 }} src={maturationImage} alt="maturation" />
        </div>
        <div className={fourthDiv}>
          <img style={{ opacity: linePoint === 3 ? 1 : 0.4 }} src={seedingImage} alt="seeding" />
        </div>
      </div>
    </div>
  );
};
BigramCircle.propTypes = {
  circlePosition: PropTypes.string.isRequired,
  linePoint: PropTypes.number.isRequired,
  isBrown: PropTypes.bool
};
BigramCircle.defaultProps = { isBrown: false };
export default BigramCircle;
