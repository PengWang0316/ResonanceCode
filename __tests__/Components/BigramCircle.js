import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import BigramCircle from '../../app/Components/BigramCircle';

describe('BigramCircle test', () => {
  const defaultProps = {
    circlePosition: 'circlePosition',
    linePoint: 0
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<BigramCircle {...props} />);

  test('BigramCircle linePoint 0 snapshot', () => expect(renderer.create(<BigramCircle {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('BigramCircle linePoint 1 snapshot', () => expect(renderer.create(<BigramCircle circlePosition="circlePosition" linePoint={1} />).toJSON()).toMatchSnapshot());

  test('BigramCircle linePoint 2 snapshot', () => expect(renderer.create(<BigramCircle circlePosition="circlePosition" linePoint={2} />).toJSON()).toMatchSnapshot());

  test('BigramCircle linePoint 3 snapshot', () => expect(renderer.create(<BigramCircle circlePosition="circlePosition" linePoint={3} />).toJSON()).toMatchSnapshot());
});
