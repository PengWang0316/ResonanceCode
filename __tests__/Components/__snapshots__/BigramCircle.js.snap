// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigramCircle test BigramCircle linePoint 0 snapshot 1`] = `
<div
  className="circle-circlePosition"
>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="firstDiv emphasizedZero"
    >
      <img
        alt="conception"
        className="ml-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="secondDiv"
    >
      <img
        alt="seeding"
        className="mr-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
  </div>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="thirdDiv"
    >
      <img
        alt="maturation"
        className="ml-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
    <div
      className="fourthDiv"
    >
      <img
        alt="growth"
        className="mr-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`BigramCircle test BigramCircle linePoint 1 snapshot 1`] = `
<div
  className="circle-circlePosition"
>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="firstDiv"
    >
      <img
        alt="conception"
        className="ml-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
    <div
      className="secondDiv"
    >
      <img
        alt="seeding"
        className="mr-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
  </div>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="thirdDiv"
    >
      <img
        alt="maturation"
        className="ml-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
    <div
      className="fourthDiv emphasizedOne"
    >
      <img
        alt="growth"
        className="mr-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`BigramCircle test BigramCircle linePoint 2 snapshot 1`] = `
<div
  className="circle-circlePosition"
>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="firstDiv cancelBottomBorder"
    >
      <img
        alt="conception"
        className="ml-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
    <div
      className="secondDiv"
    >
      <img
        alt="seeding"
        className="mr-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
  </div>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="thirdDiv emphasizedTwo"
    >
      <img
        alt="maturation"
        className="ml-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="fourthDiv cancelLeftBorder"
    >
      <img
        alt="growth"
        className="mr-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`BigramCircle test BigramCircle linePoint 3 snapshot 1`] = `
<div
  className="circle-circlePosition"
>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="firstDiv cancelRightBorder"
    >
      <img
        alt="conception"
        className="ml-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
    <div
      className="secondDiv emphasizedThree"
    >
      <img
        alt="seeding"
        className="mr-2 mt-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </div>
  </div>
  <div
    className="internalDiv d-flex"
  >
    <div
      className="thirdDiv"
    >
      <img
        alt="maturation"
        className="ml-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
    <div
      className="fourthDiv cancelTopBorder"
    >
      <img
        alt="growth"
        className="mr-2 mb-2"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "opacity": 0.4,
          }
        }
      />
    </div>
  </div>
</div>
`;
